shap/plots/_beeswarm.py

```diff
@@ -564,6 +564,9 @@ def summary_legacy(
         by default.
 
     """
+    # initialize the plot
+    pl.clf()
```

Comment:
could you please explain how this fixes the issue?

Reply:
I see. Could you please a test for this? We have tests where we compare the generated plot to a baseline one (which was generated previously). You can check that out here ([baseline plots](https://github.com/shap/shap/tree/master/tests/plots/baseline), [pytest-mpl](https://www.bing.com/search?pc=MOZI&form=MOZLBR&q=pytest+mpl), [a test of ours](https://github.com/shap/shap/blob/master/tests/plots/test_beeswarm.py#L45))

Reply:
No problem. I'll write a test for this and push it up shortly.

Reply:
Hi @CloseChoice 
I've added a test for `summary_plot`. Let me know if there's anything else you'd like me to adjust!

Reply:
thanks @chun61205. This looks great to me. Thanks a lot for your great work. Will merge when the pipelines are on green (except for the MacOS one, that is expected to fail).